ui/app/routes/vault/cluster/oidc-callback.js

```diff
@@ -5,33 +5,40 @@
 
 import Route from '@ember/routing/route';
 
+export function getParamsForCallback(qp, searchString) {
+  const queryString = decodeURIComponent(searchString);
+  let { path, code, state, namespace } = qp;
+  // namespace from state takes precedence over the cluster's ns
+  if (state?.includes(',ns=')) {
+    // TODO: does it ever get here, since Ember params strips out `=` and after?
```

Comment:
The `state` returned from paramsFor only has the format `st_yOarDguU848w5YZuotLs,ns` when the namespace portion is unencoded, which so far we've only seen with the AD FS callback uri (so it doesn't necessarily strip out following `=`

 I wrote a comment with examples [here](https://github.com/hashicorp/vault/blob/main/ui/tests/unit/routes/vault/cluster/oidc-callback-test.js#L183-L192)

Reply:
Thanks, I tested locally and found when it does this!